example/index.js

```diff
@@ -34,28 +34,26 @@ window.userale.options({
  * Note that for surgical filters, you may need to clear or reset back to a global filter callback
  * the same is true for the 'map' API. See examples below:
  */
-window.userale.filter(function (log) {
-    var type_array = ['mouseup', 'mouseover', 'mousedown', 'keydown', 'dblclick', 'blur', 'focus', 'input', 'wheel'];
-    var logType_array = ['interval'];
-    return !type_array.includes(log.type) && !logType_array.includes(log.logType);
+window.userale.addCallbacks({
+    filter(log) {
+        var type_array = ['mouseup', 'mouseover', 'mousedown', 'keydown', 'dblclick', 'blur', 'focus', 'input', 'wheel'];
+        var logType_array = ['interval'];
+        return !type_array.includes(log.type) && !logType_array.includes(log.logType);
+    }
 });
 
 /**Log Mapping API
  *
  * the 'map' API allows you to add or modify new fields to your logs
  * this example works with the "Click Me!" button at the top of index.html
  */
-document.addEventListener('click', function(e){
-    if (e.target.innerHTML === 'Click Me!') {
-        window.userale.map(function (log) {
+window.userale.addCallbacks({
+    map(log, e) {
+        if(e && e.type === 'click' && e.target.innerHTML === 'Click Me!') {
```

Comment:
Correct me if I'm wrong here, but isn't `document.addEventListener('click', ...` doing something fundamentally different than `userale.addCallbacks`? The former is a browser function while the latter is a userale function. Does this mean that the callbacks in `userale.addCallbacks` will be exected at every event?

Reply:
Code cleanup is one thing to change ... but don't forget about potential users.  If a log message will not be produced, how likely is that to effect consumers?  I don't think that seems to be something we guarantee, but could be considered a 'breaking' change, so things like that warrant at least consideration [ OK to change, but important to consider ].

Reply:
I changed the example page to match the old version exactly. And fixed some other bugs that I discovered during the process.

Reply:
Very nice work here. Couple of additional suggestions:
1. If we're all in agreement that `addCallback` is a better fit than `map` and `filter`, add deprecation flags to the docs on those functions saying they will be removed in a future release
2. Add/update a journey test

Reply:
sorry for the delay in jumping in, but I fully support these changes. I realize that this could be a breaking change down the line, but this is long overdue and eventually this kind of thing needs to happen. Fully agree with @UncleGedd's do-outs. After that's done, I see no issues moving this to test and then merging with Master if unit and some integration tests pass. If @UncleGedd and/or @brucearctor want to merge with test, I can help run an integration test this weekend, but definitely don't want to be a blocker if no one else sees any!